internal/alertmanager/upstream.go

```diff
@@ -10,12 +10,15 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// Option allows to pass functional options to NewAlertmanager()
+type Option func(am *Alertmanager)
+
 var (
 	upstreams = map[string]*Alertmanager{}
 )
 
 // NewAlertmanager creates a new Alertmanager instance
-func NewAlertmanager(name, uri string, timeout time.Duration) error {
+func NewAlertmanager(name, uri string, opts ...Option) error {
```

Comment:
nit: I'd argue this should be `AddAlertmanager` because while you do allocate you don't actually get the Alertmanager instance back.

For testing reasons (you're creating the Alertmanger manually in test) perhaps it should, and leave it up to the consumer to add it to the upstreams slice/type. I'll accept this as a separate PR though.

Reply:
Yeah, I'm refactoring this